test/Errors.test.js

```diff
@@ -2,357 +2,682 @@
+		} catch (err) {
+			// capture sync throwm errors
+			reject(err);
+		}
+	});
+
+	expect(typeof stats).toEqual("object");
+	const statsResult = stats.toJson({ errorDetails: false });
+	expect(typeof statsResult).toBe("object");
+	const { errors, warnings } = statsResult;
+	expect(Array.isArray(errors)).toBe(true);
+	expect(Array.isArray(warnings)).toBe(true);
+
+	return { errors, warnings };
+}
+
+it("should emit warning for missingFile", async () => {
+	await expect(
+		compile({
+			entry: "./missingFile"
+		})
+	).resolves.toMatchInlineSnapshot(`
+Object {
+  "errors": Array [
+    Object {
+      "loc": "4:0-20",
+      "message": "Module not found: Error: Can't resolve './missing' in '<cwd>/test/fixtures/errors'",
+      "moduleId": 0,
+      "moduleIdentifier": "<cwd>/test/fixtures/errors/missingFile.js",
+      "moduleName": "./missingFile.js",
+      "moduleTrace": Array [],
+    },
+    Object {
+      "loc": "12:9-34",
+      "message": "Module not found: Error: Can't resolve './dir/missing2' in '<cwd>/test/fixtures/errors'",
+      "moduleId": 0,
+      "moduleIdentifier": "<cwd>/test/fixtures/errors/missingFile.js",
+      "moduleName": "./missingFile.js",
+      "moduleTrace": Array [],
+    },
+  ],
+  "warnings": Array [],
+}
+`);
+});
+
+it("should emit warning for require.extensions", async () => {
+	await expect(compile({ entry: "./require.extensions" })).resolves
+		.toMatchInlineSnapshot(`
+Object {
+  "errors": Array [],
+  "warnings": Array [
+    Object {
+      "loc": "1:0-18",
+      "message": "require.extensions is not supported by webpack. Use a loader instead.",
+      "moduleId": 0,
+      "moduleIdentifier": "<cwd>/test/fixtures/errors/require.extensions.js",
+      "moduleName": "./require.extensions.js",
+      "moduleTrace": Array [],
+    },
+  ],
+}
+`);
+});
+
+it("should emit warning for require.main.require", async () => {
+	await expect(compile({ entry: "./require.main.require" })).resolves
+		.toMatchInlineSnapshot(`
+Object {
+  "errors": Array [],
+  "warnings": Array [
+    Object {
+      "loc": "1:0-20",
+      "message": "require.main.require is not supported by webpack.",
+      "moduleId": 0,
+      "moduleIdentifier": "<cwd>/test/fixtures/errors/require.main.require.js",
+      "moduleName": "./require.main.require.js",
+      "moduleTrace": Array [],
+    },
+  ],
+}
+`);
+});
+it("should emit warning for module.parent.require", async () => {
+	await expect(compile({ entry: "./module.parent.require" })).resolves
+		.toMatchInlineSnapshot(`
+Object {
+  "errors": Array [],
+  "warnings": Array [
+    Object {
+      "loc": "1:0-21",
+      "message": "module.parent.require is not supported by webpack.",
+      "moduleId": 0,
+      "moduleIdentifier": "<cwd>/test/fixtures/errors/module.parent.require.js",
+      "moduleName": "./module.parent.require.js",
+      "moduleTrace": Array [],
+    },
+  ],
+}
+`);
+});
+
+if (fs.existsSync(path.resolve(__dirname, "fixtures/errors/FILE.js"))) {
```

Comment:
Personally I'd extract this into a variable or a local helper function - or maybe just add a comment, so it's clear that the reason for this is to check for case sensitivity on the filesystem

Reply:
eg

```javascript
const isCaseSensitiveFilesystem =  fs.existsSync(path.resolve(__dirname, "fixtures/errors/FILE.js"));
if (isCaseSensitiveFilesystem) {
```

Reply:
would be `isCasePreservedFilesystem`, but i agree = ill change it.